modules/js/rackandpinion.js

```diff
@@ -1,680 +1,913 @@
-    }
-  }
-  if(paired == true){
-    if (flipY == true){
-      changeBody(compositeArray.length-4)
-      compositeArray[1].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.5)-(radius+toothHeight/2)
-      compositeArray[0].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.5)-(2*radius+toothHeight)-toothHeight-5
-      compositeArray[compositeArray.length-3].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.5)+(2*radius+toothHeight)+toothHeight+5
-      compositeArray[compositeArray.length-4].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.5)+(radius)+(toothHeight/2)
-      compositeArray[compositeArray.length-4].rotation = Math.PI
-      Body.setAngle(compositeArray[compositeArray.length-4].bodies[0], Math.PI);
-      compositeArray[1].rotation = 0
-    }
-    else{
-      changeBody(compositeArray.length-2)
-      compositeArray[1].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.5)-(radius+toothHeight/2)
-      compositeArray[0].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.5)-(2*radius+toothHeight)-toothHeight-5
-      compositeArray[compositeArray.length-1].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.5)+(2*radius+toothHeight)+toothHeight+5
-      compositeArray[compositeArray.length-2].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.5)+(radius)+(toothHeight/2)
-      compositeArray[compositeArray.length-2].rotation = Math.PI
-      Body.setAngle(compositeArray[compositeArray.length-2].bodies[0], Math.PI);
-      compositeArray[1].rotation = 0
-    }
-  }
-  
-}
-function largeGear(){
-  angleFactor = 0.35
-  deleteConstraint(compositeArray[1].bodies[0], compositeArray[0].bodies[0])
-  Body.setPosition(compositeArray[0].bodies[0], {x:compositeArray[0].constraints[0].pointA.x, y:compositeArray[0].constraints[0].pointA.y+130} )
-  Body.setAngle(compositeArray[1].bodies[0], 0)
-  radius = 80;
-  compositeArray[1].radius = radius
-  steps = (0.25 * radius)*2;
-  //toothHeight = 20;
-  toothWidthDegree = 2;
-  toothWidth = (toothWidthDegree/conversionFactor);
-  if(compositeArray[1].alternate == false){
-    changeBodyContinuous(1)
-  }
-  else{
+  toothWidth = toothWidthDegree / conversionFactor;
+  if (compositeArray[1].alternate == false) {
+    changeBodyContinuous(1);
+  } else {
     changeBody(1);
   }
-  compositeArray[1].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.45)+(radius+(toothHeight*1.8))
-  if(shared == true){
-    if (flipY == true){
-      compositeArray[compositeArray.length-3].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.45)+(radius*2)+(toothHeight*2)+25
-    }
-    else{
-      compositeArray[compositeArray.length-1].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.45)+(radius*2)+(toothHeight*2)+25
+  compositeArray[1].constraints[0].pointA.x =
+    window.innerWidth * (0.75 * 0.45) + (radius + toothHeight * 1.8);
+  if (shared == true) {
```

Comment:
Strict equality is more of what is used (unless we intentionally want to loosely compare) in modern JS:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Strict_equality

so
`if (shared === true) {`